### PR TITLE
Add MSVC-internal testing scripts to public GitHub.

### PR DESCRIPTION
**Pull request by BillyONeal**
This adds the scripts for the former "vcpkg unstable" run we use to test the compiler to GitHub to onboard them to the new world order Azure Pipelines YAML infrastructure.

### CODE REVIEW

File: scripts/azure-pipelines/windows-unstable/rearrange-msvc-drop-layout.ps1 [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+$toolsets = Get-ChildItem -Path $MSVCRoot -Directory | Sort-Object -Descending
+if ($toolsets.Length -eq 0) {
+    throw "Could not find Visual Studio toolset!"
+}
+
+Write-Host "Found toolsets:`n$($toolsets -join `"`n`")`n"
+$selectedToolset = $toolsets[0]
+Write-Host "Using toolset: $selectedToolset"
+for ($idx = 1; $idx -lt $toolsets.Length; $idx++) {
```
Comment (ras0219) [inline scripts/azure-pipelines/windows-unstable/rearrange-msvc-drop-layout.ps1:65]: This procedure may not be correct -- if I recall correctly there have been cases where the libs are stored in a differently versioned folder than the compiler.

However, I think it's fine to deal with that if/when it occurs than to worry about it right now.

File: scripts/azure-pipelines/windows-unstable/rearrange-msvc-drop-layout.ps1 [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    $badToolset = $toolsets[$idx]
+    Write-Host "Deleting toolset: $badToolset"
+    Remove-Item $badToolset -Recurse -Force
+}
+
+Write-Host "Deploying $tempRoot => $selectedToolset"
+Copy-Item "$tempRoot\*" $selectedToolset -Recurse -Force
```
Comment (ras0219) [inline scripts/azure-pipelines/windows-unstable/rearrange-msvc-drop-layout.ps1:72]: `Move-Item`?

**Review by ras0219 — APPROVED** [review]

File: scripts/azure-pipelines/windows-unstable/rearrange-msvc-drop-layout.ps1 (hunk 1 above)
Comment (BillyONeal) [reply scripts/azure-pipelines/windows-unstable/rearrange-msvc-drop-layout.ps1:65]: Technically this should also be overriding the ones in system32 etc. for maximum correctness but the old infrastructure didn't do that and I'm preserving that here.

File: scripts/azure-pipelines/windows-unstable/rearrange-msvc-drop-layout.ps1 (hunk 2 above)
Comment (BillyONeal) [reply scripts/azure-pipelines/windows-unstable/rearrange-msvc-drop-layout.ps1:72]: This does copy followed by delete because it crosses drives

**Pull request merged**